get-artifacts/get_artifacts.sh

```diff
@@ -0,0 +1,168 @@
+  echoerr -d target directory where subdirectories for jobs will be created and artifacts will be unzip to.
+  echoerr -i the single job to artifacts from.  Overrides -b,-w, and -h
+  echoerr -z decompress and delete the original downloaded artifacts?
+  echoerr -? this message.
+  echoerr output will written in to the -t target directory.
+}
+
+TOKEN=
+HISTORY=
+ARTIFACTS=
+REPO=
+WORKFLOW=
+BRANCH=
+TARGET_DIR=
+WORKFLOW_RUN_ID=
+DECOMPRESS=false
+
+
+while getopts 'h:a:r:w:b:i:t:d:z' OPTION; do
+  case "$OPTION" in
+    h)
+      HISTORY="$OPTARG"
+      ;;
+    a)
+      IFS=' ' read -ra ARTIFACTS <<< "$OPTARG"
+      ;;
+    r)
+      REPO="$OPTARG"
+      ;;
+    w)
+      WORKFLOW="$OPTARG"
+      ;;
+    b)
+      BRANCH="$OPTARG"
+      ;;
+    i)
+      WORKFLOW_RUN_ID="$OPTARG"
+      ;;
+    d)
+      TARGET_DIR="$OPTARG"
+      ;;
+    z)
+      DECOMPRESS="true"
+      ;;
+    t)
+      TOKEN="$OPTARG"
+      ;;
+    ?)
+      usage
+      exit 1
+      ;;
+  esac
+done
+
+if [[ -z "${TARGET_DIR}" ]]; then
+  echoerr target directory must be sepecified.
+  usage
+  exit 1
+fi
+
+if [[ -z "${WORKFLOW_RUN_ID}" ]]; then
+  if [[ -z "${BRANCH}" ]] || [[ -z "${WORKFLOW}" ]]; then
+    echoerr 'You must either specify a workflow_run_id (-i), or a branch (-b) and a workflow file (-w).'
+    usage
+    exit 1
+  fi
+fi
+
+# Retry seems to add a lot of time overhead.  Need to look in to this more.
+# curl_attri=("--retry" "3")
+curl_attri=('-H' 'Accept: application/vnd.github.v3+json')
+if [ -n "$TOKEN" ]; then
+  curl_attri+=('-H' "authorization: Bearer ${TOKEN}")
+fi
+echoerr Curl Parameters: "${curl_attri[@]}"
+
+
+# Conditionally retrieves the artifacts from a single jobs, the input parameter which must correspond to a numeric workflow_run_id in github actions
+# and stores the desired artifacts, contained in a list called "ARTIFACTS", in a subdirectory of TARGET_DIR, equal to the workflow_run_id/artifact_name.
+# Artifacts are always zips.   The zip file is downloaded and optionally unziped in to the job's sub directory in the TARGET_DIR, and then deleted.
+# Should the <job dir>/<artifact_name> already exist, no downloads are attempted, as the assumption would be this function has already populated the
+# artifacts in to the job dir.
+#
+# INPUT:
+#  Parameter 1:  A number workflow_run_id
+#
+# ENVIRONMENT:
+#  TARGET_DIR: A root dir for creation of sub directories for each job
+#  ARTIFACTS: A bash array of artifact names to download from github artifacts workflows for this job.
+#  REPO: Github slug corresponding to a repository
+#  UNZIP: unzip and delete the downloaded artifact?
+#
+# SIDE EFFECTS:
+#  The ${TARGET_DIR}/${Parameter 1} dir id populated with the unzipped contents of the artifacts from the job.
+#
+getArtifactsFromJob() {
+  jobId=$1;
+  echoerr getting artifacts for jobId "$jobId"
+  artifact_info="$(curl "${curl_attri[@]}" "https://api.github.com/repos/${REPO}/actions/runs/${jobId}/artifacts"  2>/dev/null)"
```

Comment:
What if an error happens here? The error message will be eaten up I believe.

Reply:
There are lots of different types of errors, the most important ones are actually login failures, which will result in a json message being returned about incorrect logins.   Regardless as long as anything is returned, and jq can't find the artifacts, the entire returned message is printed.   If it's null, there likely wont be much usable in the curl output anyway, at least that's been my experience, and argument could be made to capture it and display it with the artifact_info below... but again, just hasn't been useful.

Reply:
actually was a good call --silent (-s) and --show-error (-S) are now used everywhere.